base/math.jl

```diff
@@ -537,54 +543,56 @@ Stacktrace:
 [...]
 ```
 """
-hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-function hypot(x::T, y::T) where T<:Number
-    ax = abs(x)
-    ay = abs(y)
-    if ax < ay
-        ax, ay = ay, ax
+hypot(x::Number, y::Number; kwargs...) = hypot(promote(x, y)...; kwargs...)
+hypot(x::Complex, y::Complex; kwargs...) = hypot(promote(abs(x),abs(y))...; kwargs...)
+hypot(x::Integer, y::Integer; kwargs...) = hypot(promote(float(x), float(y))...; kwargs...)
+function hypot(x::T,y::T; correctlyrounded::Bool=false) where T<:AbstractFloat
```

Comment:
Do we really need to add a `correctlyrounded` keyword arg?  I think it's better not to add this, since we don't do this with other math functions.

I'm in favor of just using `Base.Math.FMA_NATIVE`  conditional and otherwise the 'almost' correctly rounded version, if not `FMA_NATIVE`

Reply:
Great point. I'll remove that.